### PR TITLE
Feature: cli: default download; improved REAME and documentation

### LINKED ISSUE #903: [BUG] CLI: pytube -c https://url-to-youtube does not list available language codes
**Describe the bug**

CLI: pytube -c https://url-to-youtube does not list available language codes as described in help output pytube --help

**To Reproduce**
This command does not show the language codes:
pytube -c https://www.youtube.com/watch?v=Sm4q-gtUH7Q&t=513s

This command **does**  show the language codes:
pytube -c -v https://www.youtube.com/watch?v=Sm4q-gtUH7Q&t=513s

- The code where the problem is occurring. 
_parse_args() in cli.py

**Expected behavior**

pytube -c https://www.youtube.com/watch?v=Sm4q-gtUH7Q&t=513s
Should give:
Available caption codes are: nl.3qLcwtbWM-Y, en.eEY6OEpapPo, a.fr, it._0blXBczMBo, pt-PT.rv3jYsRfesk

**Output**
pytube help output

**System information**
Please provide the following information:
- Python version (run `python --version`)
python 3.6.9, 3.7 and 3.8
- Pytube version (run `print(pytube.__version__)` in python)
4.10.1
- Command used to install pytube
pip install .

### PR DESCRIPTION
**Pull request by yvainyork**
The following features have been added:
- default download behavior for more convenience / simpler usability:
  ```pytube https://www.youtube.com/watch?my_video```
  simply downloads a the best video/audio stream. (see #901 )
- logging to a file in DEBUG level
- documentation in README.md and commandline_interface.rst added

### CODE REVIEW

File: pytube/version.py
Comment (tfdahlin) [inline pytube/version.py]: We don't update the version number in pull requests, this is handled with bump2version after a PR is merged, so please revert this.

File: pytube/helpers.py
Comment (tfdahlin) [inline pytube/helpers.py]: I don't think we should default to logging to a file, especially since this is a breaking change. Maybe make the log_filename optional, and if it exists, have setup_logger log to a file instead of stdout.

File: docs/commandline_interface.rst
Comment (tfdahlin) [inline docs/commandline_interface.rst]: Pytube should be capitalized since it's the start of a sentence, CLI should also be in caps

File: docs/commandline_interface.rst
Comment (tfdahlin) [inline docs/commandline_interface.rst]: I think this file should actually be moved to be more in-line with other documentation:
`docs/commandline_interface.rst` -> `docs/user/cli.rst`
This will also require a few changes, including adding `.. _cli:` to the start of the file, and `docs/index.rst` will need to have the file added to its User Guide section

File: README.md
Comment (tfdahlin) [inline README.md]: For consistency, please use `http://youtube.com/watch?v=2lAe1cqCOXo` as your video example.

File: README.md
Comment (tfdahlin) [inline README.md]: Looks like you made an incomplete wording change here, maybe you meant `To download a specific subtitle [...]`?

File: pytube/cli.py
Comment (tfdahlin) [inline pytube/cli.py]: I think we can do this a better way, especially since this will cause the error to read funny if a stream isn't found ("Could not find a stream with resolution: best").

Comment (tfdahlin) [reply pytube/cli.py]: Notably, there's a conflict with the `--ffmpeg` [argument](https://github.com/pytube/pytube/blob/master/pytube/cli.py#L152) which downloads and merges the highest quality video and audio streams.

**Review by tfdahlin — CHANGES_REQUESTED** [review]

File: pytube/version.py
Comment (yvainyork) [reply pytube/version.py]: I see, no problem :)

File: pytube/helpers.py
Comment (yvainyork) [reply pytube/helpers.py]: I was also considering that - I think, making it optional is the least breaking change ...

File: docs/commandline_interface.rst
Comment (yvainyork) [reply docs/commandline_interface.rst]: I just put it there, because I had no better idea, where to put it
I will move it in user - do you think, that all users know, what  '_CLI_' means (I tend to  avoid abbreviations for clarity) ? 
(since there will  link in the documentation it might be not too critical)

File: pytube/cli.py
Comment (yvainyork) [reply pytube/cli.py]: > I think we can do this a better way, especially since this will cause the error to read funny if a stream isn't found ("Could not find a stream with resolution: best").

will such an Error really occur since  youtube.streams.get_highest_resolution() should always provide a video stream (what ever the best stream is)  ? or is there a case thinkable, where you youtube will not provide a video stream with some resolution ?

Comment (yvainyork) [reply pytube/cli.py]: > Notably, there's a conflict with the `--ffmpeg` [argument](https://github.com/pytube/pytube/blob/master/pytube/cli.py#L152) which downloads and merges the highest quality video and audio streams.

You mean that _ffmepg_downloader() would be also triggered by this ? then the command line arguments are not well designed ...

**Comment by yvainyork** [comment]
Hi @tfdahlin, I think I fixed most of your proposals now ... the missing parts I need to change, after I got some sleep ...

File: pytube/cli.py
Comment (tfdahlin) [reply pytube/cli.py]: You're right, it *shouldn't* cause that error to happen now that I think about it. When I wrote this comment originally, I meant to follow up on it a bit. I think offering the "best" resolution by default isn't a great idea, which I talk about in more detail [here](https://github.com/pytube/pytube/issues/901#issuecomment-753535009). 

We should probably spend a bit more time with this PR before we merge it to iron this out. In a lot of cases, the "best" resolution provided by get_highest_resolution() isn't actually the best possible because of adaptive streams. The `-f` flag will provide the highest quality in most cases, but introduces an external dependency, `ffmpeg`.

**Comment by tfdahlin** [comment]
No worries at all! After you make the changes you want to make, tag me and I'll take another look at it. I may also make some changes myself to fix some other issues when I get the time to.

File: pytube/cli.py
Comment (yvainyork) [reply pytube/cli.py]: Regarding the "best" resolution, I made some changes that avoid this issue completely. youtube.streams.get_highest_resolution() should throw an pytube.exceptions.VideoUnavailable error to be completely safe.

Switching to option 2 (output some help if no parameter is provided is also ok - but not so intuitive, since people tend to forget options) - see my comment in #901.

**Comment by yvainyork** [comment]
Hi @tfdahlin , I am done with the changes to do some basic logging to a logfile. Regarding the default behavior when no arguments are provided, the community or the maintainers should decide, what is best and then implement one of the two options.
I think, at least the new README and docs/users/cli.rst might be merged into master (just to demonstrate, what is already working)

**Comment by tfdahlin** [comment]
@yvainyork sorry it took so long to get to this. I made a few changes to hopefully address your concerns. 

1. The `-c` flag was having issues with the argparser. When you ran `pytube -c <video_url>`, it was parsing the video url as a language code, and failing to find a URL. Instead, I created a new flag `-lc` or `--list-captions`, which will list captions.
2. I made some revisions to your default behavior. It should work the same, however I made it more clear that it will download the highest quality *progressive* stream, and restructured some of the code for clarity.

**Comment by tfdahlin** [comment]
I still need to fix the tests, going to get to that now

**Comment by tfdahlin** [comment]
@RONNCC take a peek at this and tell me what you think?

File: pytube/cli.py
Comment (RONNCC) [inline pytube/cli.py]: can you make an issue corresponding to this and put the ticket number in the todo

File: pytube/cli.py
Comment (RONNCC) [inline pytube/cli.py:467]: instead of highest resolution can we do somewhere in the middle / hardcode 480 or similar..

default behavior of highest may cause frustration to users without a strong internet connection.

**Review by RONNCC — COMMENTED** [review]
"Feature: cli: default download; improved REAME and documentation #902
" -> Feature: cli: default download; improved **README** and documentation #902

also maybe add a progress bar in the future?

**Comment by RONNCC** [comment]
@tfdahlin  done // lgtm

File: pytube/cli.py
Comment (tfdahlin) [reply pytube/cli.py]: Actually, I can remove this comment since I fixed that as part of my edits

File: pytube/cli.py
Comment (tfdahlin) [reply pytube/cli.py:467]: This *technically* won't be the highest quality for most videos -- at most, it will be 720p. I'm conflicted between defaulting to lowest resolution and highest resolution, but I do think that most of the people using this to download videos are probably looking to download the best they can, unless theyre using itags to more specifically select one. Maybe we can print out a message saying it's downloading the highest quality?

**Pull request merged**
